File: psup assignment/nodejs/q2.js
const readline = require('readline');
function uppercaseCheck() {
    const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
    rl.question('Enter a character: ', (char) => {
        if (char >= 'A' && char <= 'Z') {
            console.log('The character is uppercase.');
        }
        rl.close();
    });
}

File: psup assignment/java/q2.java
import java.util.*;
class UppercaseCheck {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter a character: ");
        char ch = input.next().charAt(0);
        if (ch >= 'A' && ch <= 'Z') {
            System.out.println("The character is uppercase.");
        }
        else {
            System.out.println("The character is not uppercase.");
        }
        input.close();
    }
}